compiler/resolution/virtualDispatch.cpp

```diff
@@ -611,23 +618,18 @@ static bool isVirtualChild(FnSymbol* child, FnSymbol* parent) {
   return retval;
 }
 
-static void clearRootsAndChildren() {
-  Vec<Vec<FnSymbol*>*> rootValues;
-  Vec<Vec<FnSymbol*>*> childValues;
-
-  virtualRootsMap.get_values(rootValues);
-  virtualChildrenMap.get_values(childValues);
-
-  forv_Vec(Vec<FnSymbol*>, value, rootValues)  {
-    delete value;
-  }
-
-  forv_Vec(Vec<FnSymbol*>, value, childValues) {
+static void clearOneMap(Map<FnSymbol*, Vec<FnSymbol*>*>& map) {
+  Vec<Vec<FnSymbol*>*> values;
+  map.get_values(values);
```

Comment:
Why store the values in a temporary instead of iterating over the map and deleting the values?